src/classes/RD2_OpportunityService.cls

```diff
@@ -154,7 +155,7 @@ public with sharing class RD2_OpportunityService {
     * @param oppIds set of opportnity Ids
     * @param rdPaymentMethods is the map of Recurrding donation Ids with Payment method.
     */
-    private void updatePaymentRecordOnCreatedOpp(Set<Id> oppIds, Map<Id,String> rdPaymentMethods) {
+    private void updatePaymentRecords(Set<Id> oppIds, Map<Id,String> rdPaymentMethods) {
```

Comment:
I know the change was not made in this PR, wanted to check this with you. Is partial success expected behaviour?
`Database.SaveResult[] saveResults = Database.update(paymentForUpdates, false);`

Reply:
It can happen, and we'd want to log those errors.